src/ComplexType.php

```diff
@@ -67,19 +72,26 @@ protected function generateClass()
 
         $classBaseType = $this->getBaseTypeClass();
 
-        $this->class = new PhpClass(
+        $this->class = new ClassGenerator(
             $this->phpIdentifier,
-            false,
-            $classBaseType,
-            null,
-            false,
+            empty($this->config->get('namespaceName'))
+                ? null
+                : $this->config->get('namespaceName'),
             $this->abstract
+                ? ClassGenerator::FLAG_ABSTRACT
+                : null,
+            $classBaseType
```

Comment:
How about using the fluid interface of the `ClassGenetator` to set properties like in `implementArrayAccess()` instead of specifying all the parameters in the constructor? This could add a bit more consistency.